File: src/main/kotlin/at/fhooe/mc/wifipositioning/interfaces/PlaybackCallbackInterface.java
package at.fhooe.mc.wifipositioning.interfaces;

import at.fhooe.mc.wifipositioning.model.simulation.recording.ScannedAccessPoint;

import java.util.List;

public interface PlaybackCallbackInterface {
    public void nearestAccessPoint(List<ScannedAccessPoint> scannedAccessPoints);

    public void allAccessPoints(List<ScannedAccessPoint> scannedAccessPointList);

    public void wayPointCount(int[] wayPointCount);
}

File: src/main/kotlin/at/fhooe/mc/wifipositioning/interfaces/PlaybackCallbackInterface.kt
package at.fhooe.mc.wifipositioning.interfaces

import at.fhooe.mc.wifipositioning.model.recording.ScannedAccessPoint

/**
 * A callback interface used by a player to simulate a detection of new access points.
 */
interface PlaybackCallbackInterface {
    /**
     * Fired, when new access points are detected.
     */
    fun allAccessPoints(scannedAccessPointList: List<ScannedAccessPoint>)

    /**
     * Sets the waypoint count of the callback object.
     */
    fun wayPointCount(wayPointCount: IntArray)
}
